app/vminsert/netstorage/netstorage.go

```diff
@@ -59,12 +59,6 @@ func (sn *storageNode) push(sns []*storageNode, buf []byte, rows int) error {
 		// Fast path - the buffer is successfully sent to sn.
 		return nil
 	}
-	if *dropSamplesOnOverload && atomic.LoadUint32(&sn.isReadOnly) == 0 {
```

Comment:
This change may lead to data transfer slowdown on vmstorage node overload, since now the data is re-routed to other vmstorage nodes, which may be busy with processing their own data. In this case the call to `rerouteBuftoOtherStorageNodes()` may block for long time. This defeats the purpose of `-dropSamplesOnOverload` command-line flag.